enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/MatchAcceptanceTest.scala

```diff
@@ -29,6 +29,28 @@ import scala.collection.mutable.ArrayBuffer
 
 class MatchAcceptanceTest extends ExecutionEngineFunSuite with QueryStatisticsTestSupport with NewPlannerTestSupport {
 
+  test("reported query that should work") {
+    val result = executeWithAllPlannersAndCompatibilityMode(
+      """MATCH (account23907) WHERE (ID(account23907) = 23907)
+        |MATCH (account23907)<-[rel1:`ACCOUNT`]-(result_owners)
+        |WHERE `result_owners`:`Persona::Superhero`:`Persona`
+        |   OR `result_owners`:`Persona::Supervillian`:`Persona`
+        |RETURN result_owners""".stripMargin)
+
+    result.size should equal(0)// does not throw
+  }
+
+  test("really weird query that breaks IDP") {
+    val result = executeWithAllPlannersAndCompatibilityMode(
+      """MATCH (a)
+        |WITH a WHERE true
+        |MATCH (c), (a)-[r]->(x)
+        |WHERE (a)-->(c)
+        |RETURN *""".stripMargin)
+
+    result.size should equal(0)// does not throw
+  }
+
```

Comment:
Is this query also fixed?

```
MATCH (ts)
MATCH (k)-[:M]->(sta)
OPTIONAL MATCH (sta)<-[:N]-(p)
WITH k, ts, coalesce(p, sta) AS ab
MATCH (d:A)
WHERE d.Id = ab.OtherId
WITH k, ts, d
MATCH (ts)-[:R]->(f)
RETURN k, ts, f, d
```
If yes, can you also add a test here?

Reply:
There is a test added for this.